File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 0
    clothes_hash = {}
    for i in range(len(clothes)) :
        if clothes[i][1] in clothes_hash :
            clothes_hash[clothes[i][1]] += 1
        else :
            clothes_hash[clothes[i][1]] = 1
    count = []
    value = 1
    for i in clothes_hash.values() :
        count.append(i)
    for i in count :
        value *= i + 1
    return value - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> cloth = new HashMap<>();
        for(int i = 0; i < clothes.length; i++){ // put the clothes info into map
            String category = clothes[i][1];
            if(!cloth.containsKey(category)){
                cloth.put(category, cloth.getOrDefault(category, 0) + 1);
            }else{
                cloth.put(category, cloth.getOrDefault(category, 0) + 1);
            }
        }
        Iterator<Integer> it = cloth.values().iterator();
        
        int answer = 1;
        while(it.hasNext())
            answer *= it.next() + 1;
        
        
        return answer - 1;
    }
}